gr-fft/include/gnuradio/fft/fft.h

```diff
@@ -115,26 +113,29 @@ class FFT_API fft_complex
  */
 class FFT_API fft_real_fwd
 {
-    int d_fft_size;
+    const int d_fft_size;
     int d_nthreads;
-    float* d_inbuf;
-    gr_complex* d_outbuf;
+    volk::vector<float> d_inbuf;
+    volk::vector<gr_complex> d_outbuf;
     void* d_plan;
 
 public:
     fft_real_fwd(int fft_size, int nthreads = 1);
+    // Copy disabled due to d_plan.
+    fft_real_fwd(const fft_real_fwd&) = delete;
+    fft_real_fwd& operator=(const fft_real_fwd&) = delete;
     virtual ~fft_real_fwd();
 
     /*
      * These return pointers to buffers owned by fft_impl_fft_real_fwd
      * into which input and output take place. It's done this way in
      * order to ensure optimal alignment for SIMD instructions.
      */
-    float* get_inbuf() const { return d_inbuf; }
-    gr_complex* get_outbuf() const { return d_outbuf; }
+    float* get_inbuf() const { return const_cast<float*>(d_inbuf.data()); }
```

Comment:
const_cast is almost always cheating. In this case, I think that we should simply drop the const qualifier on the method (we're changing API already anyway). Rationale: If we want to return a writable pointer, then this function is not truly const, even if it's only assisting in mutation.